src/background/assessment-data-converter.ts

```diff
@@ -183,11 +183,20 @@ export class AssessmentDataConverter {
         return null;
     }
 
-    public generateFailureInstance(description: string): UserCapturedInstance {
-        return {
+    public generateFailureInstance(description: string, path: string, snippet: string): UserCapturedInstance {
+        const instance = {
```

Comment:
nit: we can avoid all the `if` statements just using js shortcuts:

```js
        const instance = {
            id: this.generateUID(),
            description: description !== '' && description,
            selector: path !== '' && path,
            html: snippet !== '' && snippet,
        };
```
For me, it is much more readable but totally upto you for this.

Reply:
Oh awesome I was wondering if there was a more concise way to do this. Thanks!

Reply:
Never mind. Wrong approach.